great_expectations/datasource/fluent/pandas_datasource.py

```diff
@@ -545,7 +545,8 @@ def _validate_asset_name(asset_name: Optional[str] = None) -> str:
 
     def _get_validator(self, asset: _PandasDataAsset) -> Validator:
         batch_request: BatchRequest = asset.build_batch_request()
-        return self._data_context.get_validator(batch_request=batch_request)
+        # TODO: raise error if `_data_context` not set
+        return self._data_context.get_validator(batch_request=batch_request)  # type: ignore[union-attr] # self._data_context must be set
```

Comment:
Maybe `_data_context` should be a property or we have a private `get_validator` method on this datasource that will raise an error for us in a consistent way.

Reply:
That sounds like a good idea but I'd like to defer implementing that in a different PR.
This PR doesn't make this particular error any more or less likely, it just makes mypy better understand this issue.